fvm/errors/errors.go

```diff
@@ -114,18 +113,6 @@ func HandleRuntimeError(err error) error {
 		return NewUnknownFailure(err)
 	}
 
-	// External errors are reported by the runtime but originate from the VM.
-	// External errors may be fatal or non-fatal, so additional handling by SplitErrorTypes
-	if externalErr, ok := errors.GetExternalError(err); ok {
```

Comment:
I think `GetExternalError` is no longer doing anything useful, and can be deleted (from cadence) @onflow/cadence

Reply:
Not sure here, I'm not familiar with it

Reply:
Cadence can still return `ExternalErrors` (e.g: atree related non-user errors), and they should be treated as fatal errors here I guess. So maybe we should keep this?

Reply:
To add more context: With https://github.com/onflow/cadence/pull/2286, `externalErr.Recovered` is always an `error`, so this will never return a `NewUnknownFailure(externalErr)`.

So the only difference this snippet makes is to whether wrap the error with `NewCadenceRuntimeError` or not. I'm not sure what difference it makes to the FVM on wrapping/not-wrapping. My gut feeling is that `ExternalError`s, like atree/fvm errors, should not be wrapped as a Cadence errors. But totally up to the FVM.

Reply:
What [get external error](https://github.com/onflow/cadence/blob/158f50017f319875180512aeb6d2650e0fa72b1b/runtime/errors/errors.go#L215) is doing is just manually unwrapping an external error:

```
// GetExternalError returns the ExternalError in the error chain, if any
func GetExternalError(err error) (ExternalError, bool) {
	switch err := err.(type) {
	case ExternalError:
		return err, true
	case xerrors.Wrapper:
		return GetExternalError(err.Unwrap())
	default:
		return ExternalError{}, false
	}
}
```

That is why I think its no longer needed.

Reply:
Had an offline chat with Janez, think it should be OK to remove